docs/guides/building/cross-platform.md

```diff
@@ -149,10 +151,104 @@ env:
+
+Since Tauri officially only supports the msvc Windows target and not the gnu one, the setup is a bit more involved.
+
+First, make sure all your Tauri dependencies are at least version 1.3, check out the [dependency update guide] if you're not sure how.
+
+#### Install NSIS
+
+Some Linux distributions have NSIS available in their repositories, for example on Ubuntu you can install NSIS by running this command:
+
+```sh title=Ubuntu
+sudo apt install nsis
+```
+
+But on many other distributions you have to compile NSIS yourself or download Stubs and Plugins manually that weren't included in the distro's binary package. Fedora for example only provides the binary but not the Stubs and Plugins:
+
+```sh title=Fedora
+sudo dnf in mingw64-nsis
+wget https://github.com/tauri-apps/binary-releases/releases/download/nsis-3/nsis-3.zip
+unzip nsis-3.zip
+sudo cp nsis-3.08/Stubs/* /usr/share/nsis/Stubs/
+sudo cp -r nsis-3.08/Plugins/** /usr/share/nsis/Plugins/
+```
+
+On macOS you will need [Homebrew] to install NSIS:
+
+```sh title=macOS
+brew install nsis
+```
+
+#### Install LLVM and the LLD Linker
+
+Since the default Microsoft linker only works on Windows we will also need to install a new linker. To compile the Windows Resource file which is used for setting the app icon among other things we will also need the `llvm-rc` binary which is part of the LLVM project.
+
+```sh title="Ubuntu"
+sudo apt install lld llvm
+```
+
+```sh title=macOS
+brew install llvm
+```
+
+On macOS you also have to add `/opt/homebrew/opt/llvm/bin` to your `$PATH` as suggested in the install output.
+
+#### Install the Windows Rust target
+
+Assuming you're building for 64-bit Windows systems:
+
+```sh
+rustup target add x86_64-pc-windows-msvc
+```
+
+#### Install the Windows SDKs
+
+To get the Windows SDKs required by the msvc target we will use the [xwin] project:
+
+```sh
+cargo install xwin
+```
+
+Then you can use the `xwin` CLI to install the needed files to a location of your choice. Remember the location, we will need it in the next step. In this guide we will create a `xwin` directory in the Home directory.
+
+```sh
+xwin --accept-license splat --disable-symlinks --output ~/xwin
+```
+
+If you want to read Microsoft's license upon installation, you can run the same command without the `--accept-license` flag.
+
+Now, to make the Rust compiler use these files, you first have to create a `.cargo` directory in your project and create a `config.toml` file in it with the following content. Make sure to change the paths accordingly.
+
+```toml title=.cargo/config.toml
+[target.x86_64-pc-windows-msvc]
+linker = "lld"
+rustflags = [
+  "-Lnative=/home/username/xwin/crt/lib/x86_64",
+  "-Lnative=/home/username/xwin/sdk/lib/um/x86_64",
+  "-Lnative=/home/username/xwin/sdk/lib/ucrt/x86_64"
+]
+```
```

Comment:
this will break building on a native Windows machine, no? maybe change this to a build alias that has these flag specified on the command line?

Reply:
Well, it will only work on the system this file is defined on, even if you're moving from Linux to Linux (unless you use the same username lol).
But a more opt-in like approach would be good, i agree. I don't really know what you mean with build alias though, do you have a link at hand? I can only think of custom profiles right now which don't support custom rustflags.

Reply:
> Well, it will only work on the system this file is defined on, even if you're moving from Linux to Linux (unless you use the same username lol).

yeah you're right, but I was more concerned about building for Windows on Windows.

We should instead mention that this file should be changed for every system and thus shouldn't be checked into git


cargo aliases: https://doc.rust-lang.org/cargo/reference/config.html#alias

Reply:
Ah that alias thing. How would you combine that with the npm cli though?

Anyway, we can also just figure that out later, i tried to discourage this approach in the docs anyway so if users _do_ use it it's fine if it's not perfect imo.

I'll add a note about the git check-in thingy though, good call.

Reply:
cool